oskari-geojson/src/main/java/org/oskari/geojson/GeoJSONSchemaDetector.java

```diff
@@ -20,11 +20,37 @@
 
 public class GeoJSONSchemaDetector {
 
-    @SuppressWarnings("unchecked")
     public static SimpleFeatureType getSchema(Map<String, Object> json, CoordinateReferenceSystem crs) {
+        return getSchema(json, crs, false);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static SimpleFeatureType getSchema(Map<String, Object> json, CoordinateReferenceSystem crs, boolean ignoreGeometryProperties) {
         // FIXME: This creates a side-effect by modifying the input as well as returning the SimpleFeatureType.
         // Might cause problems later on...
-        GeoJSONReader2.replaceMapsWithGeometries(json);
+
+        // Map feature.geometry fields to JTS Geometries
+        replaceGeometry(json, GeoJSONReader2::toGeometry);
+        if (ignoreGeometryProperties) {
+            // Remove properties of type Map<String, Object> that were mappable to JTS geometries, otherwise leave as is
+            replaceMapProperties(json, it -> {
+                try {
+                    GeoJSONReader2.toGeometry(it);
```

Comment:
Would this read better if the if is inside the try catch? 

    Object geom = GeoJSONReader2.toGeometry(it);
    if (ignoreGeometryProperties) {
        return null;
    }
    return geom;

Reply:
Let me try